cfme/tests/services/test_dialog_regex_validation_in_catalog.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme import test_requirements
+from cfme.services.service_catalogs import ServiceCatalogs
+from cfme.utils.appliance.implementations.ui import navigate_to
+from wait_for import wait_for
+
+pytestmark = [
+    test_requirements.service,
+    pytest.mark.tier(2)
+]
+
+
+@pytest.fixture(scope="function")
+def dialog_cat_item(appliance, request, catalog):
+    service_dialog = appliance.collections.service_dialogs
+    dialog = "dialog_{}".format(fauxfactory.gen_alphanumeric())
+    ele_name = fauxfactory.gen_alphanumeric()
+    element_data = {
+        'element_information': {
+            'ele_label': "ele_{}".format(fauxfactory.gen_alphanumeric()),
+            'ele_name': ele_name,
+            'ele_desc': fauxfactory.gen_alphanumeric(),
+            'choose_type': "Text Box"
+        },
+        'options': {
+            'validation_switch': True,
+            'validation': "^([a-z0-9]+_*)*[a-z0-9]+$"
+            }
+        }
+    if appliance.version < '5.10':
+        element_data["options"].pop("validation_switch", None)
+    sd = service_dialog.create(label=dialog, description="my dialog")
+    tab = sd.tabs.create(tab_label="tab_{}".format(fauxfactory.gen_alphanumeric()),
+                         tab_desc="my tab desc")
+    box = tab.boxes.create(box_label="box_{}".format(fauxfactory.gen_alphanumeric()),
+                           box_desc="my box desc")
+    box.elements.create(element_data=[element_data])
+    request.addfinalizer(sd.delete_if_exists)
+
+    catalog_item = appliance.collections.catalog_items.create(
+        appliance.collections.catalog_items.GENERIC,
+        name=fauxfactory.gen_alphanumeric(),
+        description="my catalog",
+        display_in=True,
+        catalog=catalog,
+        dialog=sd)
+    request.addfinalizer(catalog_item.delete)
+    return catalog_item, ele_name
+
+
+def test_dialog_element_regex_validation(appliance, dialog_cat_item):
+    """Tests Service Dialog Elements with regex validation.
+
+    Testing BZ 1518971
+
+    Polarion:
+        assignee: nansari
+        caseimportance: high
+        initialEstimate: 1/16h
+    """
+    catalog_item, ele_name = dialog_cat_item
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    view = navigate_to(service_catalogs, 'Order')
+    view.fields(ele_name).fill("!@#%&")
+    wait_for(lambda: view.submit_button.disabled, timeout=7)
```

Comment:
why are we waiting here ? 
We can just assert if submit button is disabled or not .

Reply:
@sshveta I worked with @niyazRedhat on this issue. 
- Here what happens is ```view.fields(ele_name).fill("!@#%&")```(filling value in text) and checking ```view.submit_button.disabled```(whether button is disabled or not) works simultaneously. Thus test case fails to check button is disabled or not.
- So here we need to wait for sometime to see button is disabled.

Reply:
:+1: